docs/source/llms/prompt-engineering.rst

```diff
@@ -0,0 +1,341 @@
+.. _prompt-engineering:
+
+====================================
+Prompt Engineering UI (experimental)
+====================================
+
+Starting in MLflow 2.7, the MLflow Tracking UI provides a best-in-class experience for prompt
+engineering. With no code required, you can try out multiple LLMs from the
+:ref:`AI Gateway <gateway>`, parameter configurations, and prompts to build a variety of models for
+question answering, document summarization, and beyond. Using the embedded Evaluation UI, you can
+also evaluate multiple models on a set of inputs and compare the responses to select the best one.
+Every model created with the prompt engineering UI is stored in the :ref:`MLflow Model <models>`
+format and can be deployed for batch or real time inference. All configurations (prompt templates,
+choice of LLM, parameters, etc.) are tracked as :ref:`MLflow Runs <tracking>`.
+
+.. _prompt-engineering-quickstart:
+
+Quickstart
+==========
+
+The following guide will get you started with MLflow's UI for prompt engineering.
+
+Step 1: Create an AI Gateway Completions Route
+----------------------------------------------
+To use the prompt engineering UI, you need to create one or more :ref:`AI Gateway <gateway>`
+completions :ref:`Routes <routes>`. Follow the
+:ref:`AI Gateway Quickstart guide <gateway-quickstart>` to easily create a Route in less than five
+minutes. If you already have access to an AI Gateway Route of type ``llm/v1/completions``, you can
+skip this step.
+
+Step 2: Connect the AI Gateway to your MLflow Tracking Server
+-------------------------------------------------------------
+The prompt engineering UI also requires a connection between the AI Gateway and the MLflow
+Tracking Server. To connect the AI Gateway with the MLflow Tracking Server, simply set the
+``MLFLOW_AI_GATEWAY`` environment variable in the environment where the server is running and
+restart the server. For example, if the AI Gateway is running at ``http://localhost:7000``, you
+can start an MLflow Tracking Server in a shell on your local machine and connect it to the
+AI Gateway using the :ref:`mlflow server <cli>` command as follows:
+
+.. code-block:: bash
+
+   export MLFLOW_AI_GATEWAY="http://localhost:7000"
```

Comment:
```suggestion
   export MLFLOW_GATEWAY_URI="http://localhost:7000"
```

Reply:
```suggestion
   mlflow gateway start --config-path path/to/config.yaml --port 7000
   export MLFLOW_AI_GATEWAY="http://localhost:7000"
```

I'd also add `mlflow gateway start` before `MLFLOW_GATEWAY_URI`.

Reply:
Oof, great catch. Fixed the env var and added a code block for gateway start in Step 1